pkg/resources/virtual/common/common.go

```diff
@@ -0,0 +1,122 @@
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// SummaryCache provides an interface to get a summary/relationships for an object. Implemented by the summaryCache
+// struct from pkg/summarycache
+type SummaryCache interface {
+	SummaryAndRelationship(runtime.Object) (*wranglerSummary.SummarizedObject, []summarycache.Relationship)
+}
+
+// DefaultFields produces a VirtualTransformFunc through GetTransform() that applies to all k8s types
+type DefaultFields struct {
+	Cache SummaryCache
+}
+
+// GetTransform produces the default transformation func
+func (d *DefaultFields) GetTransform() types.VirtualTransformFunc {
+	return d.transform
+}
+
+// transform implements virtual.VirtualTransformFunc, and adds reserved fields/summary
+func (d *DefaultFields) transform(obj any) (any, error) {
+	raw, isSignal, err := getUnstructured(obj)
+	if isSignal {
+		return obj, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	raw = addReservedFields(raw)
+	raw, err = addSummaryFields(raw, d.Cache)
+	if err != nil {
+		return nil, fmt.Errorf("unable to add summary fields: %w", err)
+	}
+	return raw, nil
+}
+
+// addSummaryFields adds the virtual feilds for object state.
+func addSummaryFields(raw *unstructured.Unstructured, cache SummaryCache) (*unstructured.Unstructured, error) {
+	s, relationships := cache.SummaryAndRelationship(raw)
+	if s != nil {
+		data.PutValue(raw.Object, map[string]interface{}{
+			"name":          s.State,
+			"error":         s.Error,
+			"transitioning": s.Transitioning,
+			"message":       strings.Join(s.Message, ":"),
+		}, "metadata", "state")
+
+	}
+	var rels []any
+	for _, relationship := range relationships {
+		rel, err := toMap(relationship)
+		if err != nil {
+			return nil, fmt.Errorf("unable to convert relationship to map: %w", err)
+		}
+		rels = append(rels, rel)
+	}
+	data.PutValue(raw.Object, rels, "metadata", "relationships")
+
+	//summary.NormalizeConditions(raw)
+	normalizeConditions(raw)
+	return raw, nil
+}
+
+// addReservedFields adds reserved fields from the API to the provided object. Obj must be an unstructured - returns an error otherwise
+func addReservedFields(raw *unstructured.Unstructured) *unstructured.Unstructured {
+	return addIDField(raw)
+}
+
+// addIDField adds the ID field basd on namespace/name, and moves the current id field to _id if present
+func addIDField(raw *unstructured.Unstructured) *unstructured.Unstructured {
+	objectID := raw.GetName()
+	namespace := raw.GetNamespace()
+	if namespace != "" {
+		objectID = fmt.Sprintf("%s/%s", namespace, objectID)
+	}
+	currentIDValue, ok := raw.Object["id"]
+	if ok {
+		raw.Object["_id"] = currentIDValue
+	}
+	raw.Object["id"] = objectID
+	return raw
+}
+
+func normalizeConditions(raw *unstructured.Unstructured) {
+	var (
+		obj           data.Object
+		newConditions []any
+	)
+
+	obj = raw.Object
+	for _, condition := range obj.Slice("status", "conditions") {
```

Comment:
I realize that ship sailed long ago, but `Dig` is a better name for this kind of operation than `Slice`.

Reply:
I can't change this in this PR since that function is a part of wrangler. If you'd like to propose changing that please open an issue/PR targeting that repo and we can see what we we can do.